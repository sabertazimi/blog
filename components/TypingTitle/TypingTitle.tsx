import { cx } from '@components/utils'
import { siteConfig } from '@config'
import { useTypingEffect } from '@hooks'
import { useRef } from 'react'

interface Props {
  titles?: string[]
  speed?: number
  delay?: number
  loop?: boolean
  className?: string
}

function TypingTitle({
  titles = siteConfig.landingTitles,
  speed = 60,
  delay = 600,
  loop = true,
  className,
}: Props): JSX.Element {
  const ref = useRef<HTMLSpanElement>(null)
  useTypingEffect(ref, { titles, speed, delay, loop })

  return (
    <div
      className={cx(
        'typing-title-container mx-auto my-0 h-auto w-full px-0 py-0',
        'text-center text-5xl font-extrabold lg:text-9xl',
        className,
      )}
    >
      <span role="main" ref={ref} />
    </div>
  )
}

export default TypingTitle
