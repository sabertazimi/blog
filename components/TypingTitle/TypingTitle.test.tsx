import { render, screen, waitFor } from '@utils'
import { axe } from 'jest-axe'
import TypingTitle from './TypingTitle'

describe('typingTitle', () => {
  jest.mock('typed.js')

  it('should render correctly (snapshot)', () => {
    const { container } = render(<TypingTitle />)

    expect(container).toMatchSnapshot()
  })

  it('should render accessibility guidelines (AXE)', async () => {
    const { container } = render(<TypingTitle />)

    const a11y = await axe(container)

    expect(a11y).toHaveNoViolations()
  })

  it('should work correctly', async () => {
    render(<TypingTitle />)

    await waitFor(() => {
      expect(screen.getByRole('main')).toBeInTheDocument()
    })
  })
})
