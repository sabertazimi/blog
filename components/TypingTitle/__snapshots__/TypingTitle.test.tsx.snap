// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typingTitle should render correctly (snapshot) 1`] = `
<div>
  <div
    class="typing-title-container mx-auto my-0 h-auto w-full px-0 py-0 text-center text-5xl font-extrabold lg:text-9xl"
  >
    <span
      role="main"
    />
    <span
      aria-hidden="true"
      class="typed-cursor"
    >
      |
    </span>
  </div>
</div>
`;
